packages/experimental/src/global/js/utils/unwrap-if-fragment.js

```diff
@@ -1,14 +1,31 @@
 import React from 'react';
 
+/**
+ * unwrapIfFragment - walks the passed children unwrapping fragments
+ * to create a flat array of all children
+ */
 const unwrapIfFragment = (children) => {
   const newChildArray = [];
-  const realChildren =
-    children && children.type === React.Fragment
-      ? children.props.children
-      : children;
-  for (let index in realChildren) {
-    newChildArray.push(realChildren[index]);
+
+  const isFragment = (item) => item && item.type === React.Fragment;
+
+  const addChildren = (_children) => {
+    for (let index in _children) {
```

Comment:
Do we believe in "let ... of" ? Or is there some reason to avoid? It would be nice here.

```suggestion
    for (let child of _children) {
      if (isFragment(child) {
        addChildren(child.props.children);
      } else {
        newChildArray.push(child);
      }
    }
```

Just a thought.

Reply:
Makes sense.

Reply:
Hey @lee-chase and @dcwarwick, it looks like this loop here is causing the PageHeader component to break. I suppose there needs to be another check before the loop to make sure the item passed to `addChildren()` is iterable.

![Screen Shot 2021-02-18 at 12 53 44 PM](https://user-images.githubusercontent.com/10215203/108426228-849bb080-7209-11eb-8610-6786c6b1dfe2.png)

@AlexanderMelox ran into this issue when using the PageHeader component in the most recent build version.